scripts/add_new_provider.py

```diff
@@ -10,33 +10,40 @@
 from django.core.validators import URLValidator
 import libcloud.security
 
-from urlparse import urlparse
+from urlparse import urlparse, urljoin
 
 from core.models import Provider, PlatformType, ProviderType, Identity, Group,\
     IdentityMembership, AccountProvider, Quota, ProviderInstanceAction
 from core.models import InstanceAction
+from service.driver import get_account_driver
+from atmosphere.settings import secrets
+from atmosphere import settings
 
 libcloud.security.VERIFY_SSL_CERT = False
 libcloud.security.VERIFY_SSL_CERT_STRICT = False
 KVM = PlatformType.objects.get_or_create(name='KVM')[0]
 XEN = PlatformType.objects.get_or_create(name='Xen')[0]
 openstack = ProviderType.objects.get_or_create(name='OpenStack')[0]
 
-valid_url = URLValidator()
+url_validator = URLValidator()
 
 
-def require_input(question, validate_answer=None):
+def require_input(
```

Comment:
Thanks @cdosborn I think this could be retooled to use less ages and get the same desired behavior